### PR TITLE
add test case when to_csv argument is sys.stdout

### LINKED ISSUE #21561: Test Case for to_csv to sys.stdout
#### Code Sample, a copy-pastable example if possible

```python
#! /usr/bin/env python3

import sys
import pandas as pd
import numpy as np

data = pd.DataFrame(np.random.randint(low=0, high=10, size=(5, 5)),
                    columns=['a', 'b', 'c', 'd', 'e'])

print('--------- Output ----------')
data.to_csv(sys.stdout, index=False)
print('---------------------------')

```
#### Problem description

No output to sys.stdout.

#### Expected Output
+++++++++ Output ++++++++++
a,b,c,d,e
8,7,1,3,4
6,7,0,1,2
8,0,9,4,1
1,1,9,0,4
8,1,2,9,2
+++++++++++++++++++++++++++

Output with 0.23.1

+++++++++ Output ++++++++++
+++++++++++++++++++++++++++

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-128-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.1
pytest: None
pip: 10.0.1
setuptools: 39.2.0
Cython: None
numpy: 1.14.5
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 6.4.0
sphinx: 1.7.5
patsy: None
dateutil: 2.7.3
pytz: 2018.4
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.2.2
openpyxl: 2.5.4
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.5
lxml: 4.2.1
bs4: None
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by r00ta**
- [+] closes #21561
- [+] tests added / passed
- [+] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Add new test case when to_csv argument is sys.stdout

### CODE REVIEW

File: pandas/tests/io/formats/test_to_csv.py [hunk 1]
```diff
@@ -285,3 +286,21 @@ def test_to_csv_string_array_utf8(self):
             df.to_csv(path, encoding='utf-8')
             with open(path, 'r') as f:
                 assert f.read() == expected_utf8
+
+    def test_to_csv_stdout_file(self):
+        # GH 21561
+        str_array = [{'names': ['foo', 'bar']}, {'names': ['baz', 'qux']}]
+        df = pd.DataFrame(str_array)
```
Comment (minggli) [inline pandas/tests/io/formats/test_to_csv.py]: might be simpler to:
`DataFrame([['foo', 'bar'], ['baz', 'qux']], columns=['name_1', 'name_2'])`

File: pandas/tests/io/formats/test_to_csv.py [hunk 2]
```diff
@@ -285,3 +286,21 @@ def test_to_csv_string_array_utf8(self):
...
+1,"['baz', 'qux']"
+'''
+        saved_stdout = sys.stdout
+        out = StringIO()
+        sys.stdout = out
+        df.to_csv(sys.stdout, encoding='ascii')
+        output = out.getvalue()
+        assert output == expected_ascii
+        sys.stdout = saved_stdout
+        assert sys.stdout.closed is False
```
Comment (minggli) [inline pandas/tests/io/formats/test_to_csv.py]: i would do:
assert not sys.stdout.closed

**Comment by WillAyd** [comment]
Is it possible to parametrize this as a value in an existing test rather than adding a separate test case?

**Review by minggli — COMMENTED** [review]
there appear to be an existing pandas.util.testing.capture_stdout decorator

**Comment by r00ta** [comment]
Thanks for the tips! I've updated the PR :D

**Comment by gfyoung** [comment]
@r00ta : You have a couple of linting errors:

https://travis-ci.org/pandas-dev/pandas/jobs/395098214#L2876

**Comment by jreback** [comment]
lgtm. merge on green.

File: pandas/tests/io/formats/test_to_csv.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import pytest
 from pandas import DataFrame
 from pandas.util import testing as tm
-
+from pandas.util.testing import capture_stdout
```
Comment (minggli) [inline pandas/tests/io/formats/test_to_csv.py]: would it be possible to use tm.capture_stdout rather add another import?

Comment (jreback) [reply pandas/tests/io/formats/test_to_csv.py]: good point!

**Comment by jorisvandenbossche** [comment]
The remaining failure is a linting error:

pandas/tests/io/formats/test_to_csv.py:10:1: E302 expected 2 blank lines, found 1 
pandas/tests/io/formats/test_to_csv.py:291:60: W291 trailing whitespace

**Comment by jreback** [comment]
thanks @r00ta

**Pull request merged**